### PR TITLE
[DEV-2910] Revise feast registry construction logic

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR revise existing feast registry construction logic by using Feast's apply total method.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/feast/utils/on_demand_view.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def create(
                 offline_store_info.metadata.offline_store_table_name
             ]
             if fv_source.ttl is not None:
-                ttl_seconds = fv_source.ttl.seconds
+                ttl_seconds = fv_source.ttl.total_seconds()
```
Comment (gee-senbong) [inline featurebyte/feast/utils/on_demand_view.py:117]: This is existing bug, should use `total_seconds` here.

File: tests/unit/feast/utils/test_registry_construction.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+            "batchSource": {
+                **common_batch_source,
+                "name": f"fb_entity_cust_id_fjs_1800_300_600_ttl_{catalog_id}",
+                "snowflakeOptions": {
+                    **common_snowflake_options,
+                    "table": f"fb_entity_cust_id_fjs_1800_300_600_ttl_{catalog_id}",
+                },
+            },
+            "entities": ["cust_id"],
+            "entityColumns": [{"name": "cust_id", "valueType": "STRING"}],
```
Comment (gee-senbong) [inline tests/unit/feast/utils/test_registry_construction.py:227]: Using feast apply total, entity columns injected into feature view automatically.

File: featurebyte/feast/utils/registry_construction.py [hunk 3]
```diff
@@ -414,10 +460,42 @@ def create_feast_feature_services(
...
+        online_store_config: Optional[Dict[str, Any]] = None
+        if online_store:
+            online_store_for_type = {
+                OnlineStoreType.REDIS: RedisOnlineStoreConfig,
+                OnlineStoreType.MYSQL: MySQLOnlineStoreConfig,
+            }
+            online_store_config_class = online_store_for_type[online_store.details.type]
+            online_store_config = online_store_config_class(
+                **online_store.details.dict(by_alias=True, exclude={"credential": True})
+            ).dict(by_alias=True)
```
Comment (YS-L) [inline featurebyte/feast/utils/registry_construction.py:482]: Is the online store config created without the credentials information? Another comment mentioned that feast apply will be attempting to make connection to online store, will there be an issue without the credentials?

Comment (gee-senbong) [reply featurebyte/feast/utils/registry_construction.py:482]: Good catch, will update this to use `featurebyte.feast.model.online_store.get_feast_online_store_details`.

**Review by YS-L — APPROVED** [review]

**Pull request merged**